### PR TITLE
BillingApiSpec: create new billing project with retry

### PR DESCRIPTION
**Pull request by aweng98**
depend on [workbench-lib PR 221](https://github.com/broadinstitute/workbench-libs/pull/221)

Google fails to create a new billing project sometimes. The workaround is have the test to retry up to 3 times.
[JIRA QA-515](https://broadworkbench.atlassian.net/browse/QA-515)

### CODE REVIEW

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/BillingApiSpec.scala [hunk 1]
```diff
@@ -32,99 +35,115 @@ class BillingApiSpec extends FreeSpec with BillingFixtures with MethodFixtures w
...
+        val expectedStatus = "Done"
+        val actualStatus = Submission.getSubmissionStatus(billingProjectName, workspaceName, submissionId)
+        withClue(s"Submission $billingProjectName/$workspaceName/$submissionId status should be $expectedStatus") {
+          actualStatus shouldBe expectedStatus
+        }
       }
-    }
 
+      // clean up
+      Rawls.workspaces.delete(billingProjectName, workspaceName)
```
Comment (marctalbott) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/BillingApiSpec.scala:114]: No longer checking if the billing project can be deleted?

**Comment by aweng98** [comment]
jenkins retest

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/BillingApiSpec.scala (hunk 1 above)
Comment (aweng98) [reply automation/src/test/scala/org/broadinstitute/dsde/test/api/BillingApiSpec.scala:114]: Fixed. 👍

**Comment by aweng98** [comment]
fiab-start failed.
jenkins retest

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/BillingApiSpec.scala [hunk 2]
```diff
@@ -32,99 +35,123 @@ class BillingApiSpec extends FreeSpec with BillingFixtures with MethodFixtures w
...
+        val associatedBillingAccount = Google.billing.getBillingProjectAccount(billingProjectName)
+        associatedBillingAccount shouldBe Some(ServiceTestConfig.Projects.billingAccountId)
+      }
 
-        // clean up
-        Rawls.workspaces.delete(billingProjectName, workspaceName)
-        deleteBillingProject(billingProjectName)
+      // add studentA to google billing project with USER role
+      val studentA = UserPool.chooseStudent
+      Rawls.billing.addUserToBillingProject(billingProjectName, owner.email, BillingProjectRole.Owner)
```
Comment (MatthewBemis) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/BillingApiSpec.scala]: i realize that this is how the original test was set up but it seems odd that the owner is being added to the project when they're the one creating it. regardless, it's harmless

**Comment by aweng98** [comment]
jenkins retest

**Review by MatthewBemis — APPROVED** [review]
i only did a quick scan of this but seems good

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/BillingApiSpec.scala (hunk 2 above)
Comment (aweng98) [reply automation/src/test/scala/org/broadinstitute/dsde/test/api/BillingApiSpec.scala]: 👍 removed this redundant api request.

**Review by marctalbott — APPROVED** [review]

**Review by ansingh7115 — APPROVED** [review]

**Pull request merged**